MachObfuscator/SymbolMangling/RealWordsMangler/RealWordsExportTrieMangler.swift

```diff
@@ -1,32 +1,60 @@
 import Foundation
 
 protocol RealWordsExportTrieMangling: AnyObject {
-    func mangle(trie: Trie, fillingRootLabelWith labelFill: UInt8) -> Trie
+    func mangle(trie: Trie) -> Trie
 }
 
 final class RealWordsExportTrieMangler: RealWordsExportTrieMangling {
-    private let machOViewDoomEnabled: Bool
+    private let minimumFillValue: UInt8
 
     init(machOViewDoomEnabled: Bool) {
-        self.machOViewDoomEnabled = machOViewDoomEnabled
+        minimumFillValue = machOViewDoomEnabled ? 0 : 1
     }
 
-    func mangle(trie: Trie, fillingRootLabelWith labelFill: UInt8) -> Trie {
-        var trieCopy = trie
+    func mangle(trie: Trie) -> Trie {
+        // In case of Mach-O trie binary-representation, a root trie node doesn't even have a space to store its label.
+        // That's why it is always empty in the `Trie` struct. But it feels unsafe to pass `0` for
+        // `fillingRootLabelWith`, because `0` is a cstring end marker.
+        var mutableTrie = trie
+        mutableTrie.fillRecursively(startingWithFillValue: minimumFillValue,
+                                    minimumFillValue: minimumFillValue)
+        return mutableTrie
+    }
+}
+
+private extension Trie {
+    struct FillResult {
+        var finalFillValue: UInt8
+    }
 
-        trieCopy.label = trieCopy.label.map { _ in
-            labelFill
+    @discardableResult mutating func fillRecursively(startingWithFillValue initialFillValue: UInt8,
```

Comment:
Result of `fillRecursively` is generally important and you discard it only once, for root of the `Trie`. As I understand, this is not the intended use of  `@discardableResult`.

Reply:
You are right. I will remove that and use `_ =` for the root of the trie.